numba/tests/compile_with_pycc.py

```diff
@@ -115,3 +116,11 @@ def dict_usecase(arr):
     for k, v in d.items():
         out[k] = k * v
     return out
+
+# checks for issue #6386
+@cc_nrt.export('internal_str_dict', i8(unicode_type))
+def internal_str_dict(x):
+    d = typed.Dict.empty(unicode_type,i8)
+    if(x not in d):
+        d[x] = len(d)
+    return len(d)
```

Comment:
Would be good to add another test in here where something like this https://github.com/numba/numba/issues/6386#issuecomment-724043361 is AOT compiled i.e. a function hashing a string or bytes. Then this AOT module is imported in the same process and the hash is checked to be the same as the hash of the same string/bytes as called from cpython. Then, to do the same again, but without compilation, in a subprocess to check that it is getting the hash secret from a new process correctly. If this isn't clear/you need a hand with this then please shout :)

Reply:
I think I got it: f9293cb.